File: library/gradle-plugin/src/main/java/io/mehow/laboratory/gradle/ProjectFilter.java
package io.mehow.laboratory.gradle;

import org.gradle.api.HasImplicitReceiver;
import org.gradle.api.Project;
import org.jetbrains.annotations.NotNull;

@HasImplicitReceiver // Does not work with Kotlin defined interfaces in *.kts build files.
public interface ProjectFilter {
  boolean reject(@NotNull Project project);
}

File: library/gradle-plugin/src/main/java/io/mehow/laboratory/gradle/ProjectFilter.kt
package io.mehow.laboratory.gradle

import org.gradle.api.HasImplicitReceiver
import org.gradle.api.Project

/**
 * Filter for projects that should not contribute their feature flags to the code generation
 * in a merging module.
 */
@HasImplicitReceiver
public fun interface ProjectFilter {
  /**
   * A predicate that excludes [project] from code generation contribution if it returns `true`.
   */
  public fun reject(project: Project): Boolean
}
